src/gpu/kmeans/KmMatrix/utils.cuh

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright 2018 H2O.ai, Inc.
+ * License   Apache License Version 2.0 (see LICENSE for details)
+ */
+
+#ifndef UTILS_CUH_
+#define UTILS_CUH_
+
+#include "GpuInfo.cuh"
+
+namespace H2O4GPU {
+namespace KMeans {
+
```

Comment:
These thread index helpers are not specific to kmeans and so should go in a common utilities file shared by all algorithms.